Tweak and update subsection Labels of Issue Triaging

fmease opened the pull request:
1. Mention more label categories that are of importance 
   1. Mention `PG-*` labels (→ specific project groups).
   2. Mention `L-*` labels (→ specific lints).
   3. Mention `-Z*` labels (→ specific unstable `-Z` compiler flags).
2. ~~Mark keywords bold / strong. With keyword I mean words (e.g., *team*) whose initial is used as the prefix of a label category (e.g., *T-*). https://rustc-dev-guide.rust-lang.org/contributing.html#issue-triage does this, too, which I found out after the fact.~~
3. Linkify label categories just like https://rustc-dev-guide.rust-lang.org/contributing.html#issue-triage.
4. Add three footnotes for "trivia" (can be used to form mnemonics :P).
5. Link to https://rustc-dev-guide.rust-lang.org/contributing.html#issue-triage.
6. Tiny style tweaks for consistency (periods, etc.).
7. Sort the list of important `I-*` labels by ... importance and add a two or three more labels.

NB: I haven't added a bullet point for the `B-*` labels (used on tracking issues & related to unstable language features) because they're not relevant for your everyday issue triaging in my opinion.

jieyouxu commented on src/release/issue-triaging.md:
I think it's not needed to say that `A-diagnostics` don't have `C-*` labels, I think it's probably best to leave the discretion to the people labelling the issue.

fmease replied:
I explicitly requested this note to be added in https://github.com/rust-lang/rust-forge/pull/716#discussion_r1435846415. There was a time when the issue template for diagnostic issues also added the `C-bug` label next to `A-diagnostics` which is no longer the case. I don't know what the motivation for that change was. I guess `C-bug` is now implied, its removal reduces clutter and we have `D-*` labels which are way more telling and helpful for us rustc devs. 

Initially I was against it (without speaking out) because as you probably know rustc has the policy “diagnostic issues are bugs” and `C-bug` denotes that.

I've requested this line to be added to officially encode this convention / tribal knowledge because it's not documented anywhere else as far as I know.

Note that we currently have 274 `A-diagnostic+C-bug` and 1929 `A-diagnostic-C-bug` issues open. The former is 26.6% of the later / the total. Furthermore we should remove the `C-bug` from those 274 issues imo. Can do that later with a bulk update.

fmease replied:
TL;DR: I'd like to keep it.

jieyouxu commented on src/release/issue-triaging.md:
I think we also have like operating-system + compiler, e.g. `O-windows-mvsc`, so maybe call this "target" or "target families"? It's also possible for an issue to be relevant to multiple target families. Don't know what the least confusing terminology here is.

fmease replied:
Lol, I've just noticed that I'm highlighting **p**latform which doesn't make sense :facepalm:

fmease replied:
Makes sense. I've changed the terminology to *(compile) target (family)* while highlighting the *platform (arch or OS)* as the most notable instance of it.

jieyouxu commented on src/release/issue-triaging.md:
Do `P-*` labels *only* use compiler prioritization? Do libs/libs-api/lang/rustdoc teams use different procedures?

Nilstrieb replied:
generally yes

jieyouxu commented on src/release/issue-triaging.md:
There's also `E-needs-bisection` which might be worth mentioning

fmease replied:
Added.

jieyouxu reviewed (approved):
Left some nits, but LGTM otherwise.

apiraino commented:
I've tried looking [at the rendered version](https://github.com/rust-lang/rust-forge/blob/eca55fea155aa2de200f4d3661e740a634062652/src/release/issue-triaging.md) and to be honest in the new proposed version there's so much formatting that I find it hard to visually parse the document.

<details><summary>Old version</summary>

![](https://github.com/rust-lang/rust-forge/assets/6098822/69d50c71-dc73-4cff-8547-284924fa0306)

</details> 

<details><summary>New version</summary>

![](https://github.com/rust-lang/rust-forge/assets/6098822/41b6ff69-a7c8-4b21-a6b7-848faa70d31e)

</details> 

I can't pinpoint exactly all the small things that imo make the new version less readable, they are mixed with small improvements in the wording, which I find otherwise valuable.

I don't have time or energy (or UI skills) to make a detailed review so I won't oppose merging this as it is. I wish we had someone with some a11y background for an informed opinion.

Sorry @fmease I am a minimalist :laughing:

jieyouxu commented:
> I can't pinpoint exactly all the small things that imo make the new version less readable, they are mixed with small improvements in the wording, which I find otherwise valuable.

It's probably a mix of the label prefixs in different font, color and background + bolded keywords + monospace example labels that makes it a bit visually busy. Not sure how to make it less overwhelming either.

fmease commented:
I can remove the links and the weight on the keywords again and only keep the wording tweaks. That'd be fine by me

jieyouxu commented:
> I can remove the links and the weight on the keywords again and only keep the wording tweaks. That'd be fine by me

I think it's nice *fuctionally* to have them lol

Nilstrieb commented:
mdbook is less noisy than GitHub, so the comparison isn't entirely fair. I think removing the emphasis makes sense, it's fairly obvious and not that important.

Nilstrieb reviewed (approved):
thanks, cool. i'd be fine merging it as-is but if you wanna remove the emphasis and address some more feedback first that works too. just make sure to not get lost in improvements and forget the PR, rather merge early than late :)

fmease commented:
I've addressed the review comments and removed the font weight from the keywords again but kept the links because they're quite useful imo even if they might worsen the UI slightly due to their blue color.
